### PR TITLE
Enforce Ordering in DuplicateVoteEvidence

### PR DESCRIPTION
**Pull request by AdityaSripal**
Fixes issue described in: #4143 

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: types/evidence.go [hunk 1]
```diff
@@ -209,6 +231,10 @@ func (dve *DuplicateVoteEvidence) ValidateBasic() error {
 	if err := dve.VoteB.ValidateBasic(); err != nil {
 		return fmt.Errorf("Invalid VoteB: %v", err)
 	}
+	// Enforce Votes are lexicographically sorted on blockID
+	if strings.Compare(dve.VoteA.BlockID.Key(), dve.VoteB.BlockID.Key()) >= 0 {
+		return errors.New("Duplicate Votes in invalid order")
```
Comment (melekes) [inline types/evidence.go]:
```suggestion
		return errors.New("duplicate votes in invalid order")
```

**Comment by melekes** [comment]
This also needs a CHANGELOG_PENDING.md entry.

**Comment by AdityaSripal** [comment]
@melekes Would this be considered a breaking change? I don't believe it is, since the submission of evidence is taken care of completely inside tendermint

**Comment by melekes** [comment]
> @melekes Would this be considered a breaking change? I don't believe it is, since the submission of evidence is taken care of completely inside tendermint

no, just a bugfix

File: types/evidence.go [hunk 2]
```diff
@@ -209,6 +231,10 @@ func (dve *DuplicateVoteEvidence) ValidateBasic() error {
 	if err := dve.VoteB.ValidateBasic(); err != nil {
 		return fmt.Errorf("Invalid VoteB: %v", err)
 	}
+	// Enforce Votes are lexicographically sorted on blockID
+	if strings.Compare(dve.VoteA.BlockID.Key(), dve.VoteB.BlockID.Key()) >= 0 {
```
Comment (melekes) [inline types/evidence.go]: This is actually breaking since old evidence may not be in correct order.

**Comment by ebuchman** [comment]
We should update the spec for this here: https://github.com/tendermint/tendermint/blob/master/docs/spec/blockchain/blockchain.md#evidence-1

**Comment by melekes** [comment]
@ebuchman thanks for pointing this out!

**Pull request merged**